src/main/java/losty/netatmo/NetatmoUtils.java

```diff
@@ -29,50 +31,49 @@
 
 public class NetatmoUtils {
 
-	public static List<Station> parseDevicesList(JSONObject response) {
-		List<Station> devices = new ArrayList<Station>();
+	private static final Logger log = LoggerFactory.getLogger(NetatmoUtils.class);
+
+	public static List<Station> parseStationsData(final JSONObject response) {
+		final List<Station> devices = new ArrayList<Station>();
 
 		try {
 			JSONArray JSONstations = response.getJSONObject("body").getJSONArray("devices");
 
 			for (int i = 0; i < JSONstations.length(); i++) {
-				JSONObject station = JSONstations.getJSONObject(i);
-				String name = station.getString("station_name");
-				String moduleName = station.getString("module_name");
-				String id = station.getString("_id");
-
-				Station newStation = new Station(name, id);
-				Module newModule = new Module(moduleName, id, Module.TYPE_INDOOR);
-
-				newStation.addModule(newModule);
-				devices.add(newStation);
-			}
-
-			JSONArray JSONmodules = response.getJSONObject("body").getJSONArray("modules");
-
-			for (int i = 0; i < JSONmodules.length(); i++) {
-				JSONObject module = JSONmodules.getJSONObject(i);
-				String mainDevice = module.getString("main_device");
-				String name = module.getString("module_name");
-				String id = module.getString("_id");
-				String type = module.getString("type");
-
-				Module newModule = new Module(name, id, type);
-				for (Station station : devices) {
-					if (mainDevice.equals(station.getId())) {
-						station.addModule(newModule);
-					}
+				final JSONObject station = JSONstations.getJSONObject(i);
+				final String stationId = station.getString("_id");
+				final String stationName = station.getString("station_name");
+				final String stationModuleName = station.optString("module_name", stationId);
+
+				final Station newStation = new Station(stationName, stationId);
+				final Module newStationModule = new Module(stationModuleName, stationId, Module.TYPE_INDOOR);
+
+				newStation.addModule(newStationModule);
+				
+				
+				JSONArray JSONmodules = station.getJSONArray("modules");
+
+				for (int j = 0; j < JSONmodules.length(); j++) {
+					final JSONObject module = JSONmodules.getJSONObject(j);
+					final String moduleId = module.getString("_id");
+					final String moduleName = module.optString("module_name", moduleId);
+					final String moduleType = module.getString("type");
+
+					final Module newModule = new Module(moduleName, moduleId, moduleType);
+					newStation.addModule(newModule);
 				}
+				
+				devices.add(newStation);
 			}
 		} catch (JSONException e) {
-			e.printStackTrace();
```

Comment:
Actually I'm trying to remember why I did not just pass on JSONException here - we could get rid of this whole dependency to some-whatever logger...

Reply:
Yes, that also would be an option - I just wanted to get rid of the printing to System.out since I use this lib in an OSGi envyronment where errors to System.out are spaming my terminal and messages sent via most of the available logging framework are processed and written to the appropriate log file...